lib/Doctrine/DBAL/Driver/Connection.php

```diff
@@ -39,9 +39,40 @@ public function quote(string $input) : string;
     public function exec(string $statement) : int;
 
     /**
-     * Returns the ID of the last inserted row or sequence value.
+     * Returns the ID of the last inserted row.
+     *
+     * This method returns a string representing the value of the auto-increment field from the last row inserted into
+     * the database, if any, or throws a DriverException if a value cannot be returned, in particular when:
+     *
+     * - the driver does not support identity columns;
+     * - the last statement dit not return an identity (caution: see note below).
+     *
+     * Note: if the last statement was not an INSERT to an autoincrement column, this method MAY return an ID from a
+     * previous statement. DO NOT RELY ON THIS BEHAVIOR which is driver-dependent: always use lastInsertId() right after
+     * executing an INSERT statement.
+     *
+     * @return string The last insert ID.
+     *
+     * @throws DriverException If an error occurs.
+     */
+    public function lastInsertId() : string;
+
+    /**
+     * Returns the current sequence value for the given sequence name.
+     *
+     * This method returns a string representing the current value of the sequence, or throws a DriverException if a
+     * value cannot be returned, in particular when:
+     *
+     * - the driver does not support sequences;
+     * - the sequence does not exist.
+     *
+     * @param string $name The sequence name.
+     *
+     * @return string The current sequence value.
+     *
+     * @throws DriverException If an error occurs.
      */
-    public function lastInsertId(?string $name = null) : string;
+    public function getSequenceNumber(string $name) : string;
```

Comment:
Could be added on the stable branch as part of a new interface. If that new interface is not implemented, trigger a deprecation so that we can merge both interfaces in the next major release.

Reply:
Not sure what you're expecting from me exactly here?

Reply:
I'm expecting the `getSequenceNumber()` to exist in the wrapper statement but not in the driver statements.

Reply:
Sorry, you lost me. Do you want to remove `getSequenceNumber()` from the `Driver\Connection` interface, and move it to the `AbstractPlatform`, to be consumed by the `DBAL\Connection` only?

If so, I feel like this is out of scope for this PR. Maybe we should first work on merging it as is (split in 2 methods), then work on moving `getSequenceNumber()` from `Driver\Connection`  to `AbstractPlatform` in another PR.

(Plus, I don't know if we currently have the SQL to get the sequence number on all platforms, some of them may rely on PDO's ability to return a sequence number with `lastInsertId($name)`, which we'll lose if we proceed with such a change.)

Reply:
> Sorry, you lost me. Do you want to remove `getSequenceNumber()` from the `Driver\Connection` interface, and move it to the `AbstractPlatform`, to be consumed by the `DBAL\Connection` only?

Yes. Do you think this is wrong? I'm really open to a discussion.

> If so, I feel like this is out of scope for this PR

It's not because otherwise, we'll be introducing a driver connection method that will have to be removed later.

> (Plus, I don't know if we currently have the SQL to get the sequence number on all platforms, some of them may rely on PDO's ability to return a sequence number with `lastInsertId($name)`, which we'll lose if we proceed with such a change.)

It's probably only PostgreSQL that supports sequences but we only support it via `pdo_pgsql`. Implementing `getSequenceNumber()` for PostgreSQL should be as trivial as for any other platform.

Reply:
> Yes. Do you think this is wrong? I'm really open to a discussion.

I don't have a strong opinion about this. What you're saying makes sense, but at the same time I'm personally OK with the way it is now, especially the convenience of using `lastInsertId($name)` when available, even though it's trivial to re-implement it as you said.

> It's not because otherwise, we'll be introducing a driver connection method that will have to be removed later.

I'd say that being hesitant (at least, be open to discussion) on the first point would be in favour of keeping this PR the way it is, and open a new discussion later to see if it's worth moving to AbstractPlatform. Otherwise, we're (I am) venturing into unknown territory and making the current PR less likely to be merged.

Reply:
> Yes. Do you think this is wrong?

By the way, off-topic: something that's currently *really* wrong, IMHO, is to have `Connection` implement `Driver\Connection`. It should compose, not implement.

Reply:
I agree that those are two different interfaces. Not necessarily that the one should extend the other.

Reply:
For the reference, this has been addressed in https://github.com/doctrine/dbal/pull/4159.